### DISCUSSION TITLE
Copilot for IntelliJ hangs when closing the IDE while some process is running

### DISCUSSION THREAD

**Discussion by Clashsoft** [community/community#88527] [Copilot Conversations]
### Related

This issue is so widely known that it is reported many times here and on Jetbrains' Youtrack:

- https://github.com/orgs/community/discussions/88599
- https://github.com/orgs/community/discussions/105860
- https://github.com/orgs/community/discussions/108346
- https://youtrack.jetbrains.com/issue/IDEA-343155
- https://youtrack.jetbrains.com/issue/IDEA-343537

So far the best option is downgrading to 1.4.7.4174:
- https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/462996
- See https://youtrack.jetbrains.com/issue/IDEA-343537/All-JetBrains-are-waiting-for-process-detach#focus=Comments-27-9144783.0-0

### Select Topic Area

Bug

### Body

For context: https://youtrack.jetbrains.com/issue/IDEA-343155

***What steps will reproduce the issue?***

1. Open a project
2. Launch a run configuration, e.g. npm script
3. Close the IDE window
4. Wait for process detach... forever or click Kill process

***What is the expected result?***

The process should be terminated immediately, as wanted in settings.

***What happens instead?***

A window opens that expects me to click Kill process.

![](https://youtrack.jetbrains.com/api/files/74-2171541?sign=MTcwNTcwODgwMDAwMHwxMS0zMDE2Mjh8NzQtMjE3MTU0MXxjVldObzBJZWlOemMtaC1MbENvUVhhdURnU3JmOVVLX1R2cjZlWG5HdDZjDQo&updated=1705406165380)
![](https://youtrack.jetbrains.com/api/files/74-2171536?sign=MTcwNTcwODgwMDAwMHwxMS0zMDE2Mjh8NzQtMjE3MTUzNnxBMG56NlF5U2h4LXVKOGJrc2NQLWgtTlFIM05oV3MtNGE0cjRlYVJyZm0wDQo&updated=1705406003140)

```text
IntelliJ IDEA 2023.3.2 (Ultimate Edition)
Build #IU-233.13135.103, built on December 20, 2023
Licensed to Adrian Kunz
Subscription is active until October 8, 2024.
For educational use only.
Runtime version: 17.0.9+7-b1087.9 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.2.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 12
Metal Rendering is ON
Registry:
  debugger.click.disable.breakpoints=true
  ide.experimental.ui=true
Non-Bundled Plugins:
  com.github.lppedd.idea-conventional-commit (0.22.0)
  com.vecheslav.darculaDarkerTheme (1.2.0)
  com.dekonoplyov.KeymapNationalizer (0.2.8)
  org.antlr.intellij.plugin (1.22)
  com.github.copilot (1.4.9.4300)
  com.intellij.properties.bundle.editor (233.11799.172)
  com.nasller.CodeGlancePro (1.7.8)
  String Manipulation (9.12.0)
  com.jetbrains.packagesearch.intellij-plugin (233.11799.196)
  com.wakatime.intellij.plugin (14.3.8)
  org.elasticsearch4idea (2023.3.3-233)
  mobi.hsz.idea.gitignore (4.5.2)
  com.nekofar.milad.intellij.nestjs (1.2.25)
Kotlin: 233.13135.103-IJ
```

**Comment by Clashsoft** [comment]
Here is also a thread dump:

[threaddump.txt](https://github.com/community/community/files/13961478/threaddump.txt)

**Comment by montella1507** [comment]
https://youtrack.jetbrains.com/issue/IDEA-343537 

i can confirm. Check the issue above. Copilot is blocking ALL jetbrains IDEs

**Comment by salvoravida** [comment]
any news? this bug is quite boring!

**Comment by montella1507** [comment]
The problem is GitHub Copilot plugin. Disable it or downgrade and wait for fix.

**Reply by Clashsoft** [reply]
Obviously copilot is the problem. Otherwise this wouldn't be here in the Copilot Discussion Category and the numerous issues on Youtrack would still be open. It's a shame they don't fix this.

**Reply by montella1507** [reply]
Ah, my bad. i Was not sure in which issue i am participating at ( there are several issues on github with the same cause but different subject).

**Comment by jcputney** [comment]
Also hitting this issue, can we please get an update?

**Comment by cercos** [comment]
> Obviously copilot is the problem. Otherwise this wouldn't be here in the Copilot Discussion Category and the numerous issues on Youtrack would still be open. It's a shame they don't fix this.

I cancelled my copilot subscription because of it and went with the jetbrains AI assistant in the mean time.  Hope they fix it but the AI assistant has grown on me.

**Comment by megothss** [comment]
Issue still persists. 😢
